addons/web/controllers/main.py

```diff
@@ -584,18 +616,65 @@ class GroupsTreeNode:
     build a leaf. The entire tree is built by inserting all leaves.
     """
 
-    def __init__(self, model, fields, groupby, root=None):
+    def __init__(self, model, fields, groupby, groupby_type, root=None):
         self._model = model
-        self._fields = fields
+        self._export_field_names = fields  # exported field names (e.g. 'journal_id', 'account_id/name', ...)
         self._groupby = groupby
+        self._groupby_type = groupby_type
 
         self.count = 0  # Total number of records in the subtree
-        self.aggregated_values = Counter()  # Fields aggregated values {field_name: aggregated value}
         self.children = OrderedDict()
         self.data = []  # Only leaf nodes have data
+        self.aggregated_field_names = []
+        for field_name in self._export_field_names:
+            if '/' in field_name:
+                # Currently no support of aggregated value for nested record fields
+                # e.g. line_ids/analytic_line_ids/amount
+                continue
+            field = self._model._fields[field_name]
+            if field.group_operator:
+                self.aggregated_field_names.append(field_name)
+
         if root:
             self.insert_leaf(root)
 
+    def _get_aggregate(self, field_name, data, group_operator):
+        # When exporting one2many fields, multiple data lines might be exported for one record.
+        # Blank cells of additionnal lines are filled with an empty string. This could lead to '' being
+        # aggregated with an integer or float.
+        data = (value for value in data if value != '')
+
+        aggregate_func = OPERATOR_MAPPING.get(group_operator)
+        if group_operator == 'avg':
+            return self._get_avg_aggregate(field_name, data)
+
+        if self.data:
+            return aggregate_func(data)
+        return aggregate_func((child.aggregated_values.get(field_name) for child in self.children.values()))
+
+    def _get_avg_aggregate(self, field_name, data):
+        aggregate_func = OPERATOR_MAPPING.get('sum')
```

Comment:
```suggestion
        aggregate_func = OPERATOR_MAPPING.get('avg')
```
or remove avg from OPERATOR_MAPPING